rct229/rules/section6/section6rule9.py

```diff
@@ -0,0 +1,135 @@
+from rct229.ruleset_functions.normalize_interior_lighting_schedules import normalize_interior_lighting_schedules
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all, find_one_with_field_value
+from rct229.utils.pint_utils import ZERO, pint_sum
+import functools
+
+FLOOR_AREA_LIMIT = 5000 # square foot
+
+
+class Section6Rule9(RuleDefinitionListIndexedBase):
+    """Rule 9 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
+
+    def __init__(self):
+        super(Section6Rule9, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section6Rule9.RuleSetModelInstanceRule(),
+            index_rmr="proposed",
+            id="6-9",
+            description="Proposed building is modeled with other programmable lighting controls through a 10% schedule reduction in buildings less than 5,000sq.ft.",
+            list_path="ruleset_model_instances[0]",
+            required_fields={"$": ["calendar"], "calendar": ["is_leap_year"]},
+        )
+
+    def create_data(self, context, data=None):
+        rmd_p = context.proposed
+        return {"is_leap_year": rmd_p["calendar"]["is_leap_year"]}
+
+    class RuleSetModelInstanceRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule9.RuleSetModelInstanceRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule(),
+                index_rmr="proposed",
+                list_path="buildings[*]",
+                required_fields={"$": ["schedules"]},
+            )
+
+        def create_data(self, context, data=None):
+            rmd_p = context.proposed
+            return {
+                **data,
+                "schedules_p": rmd_p["schedules"],
+            }
+
+        class BuildingRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section6Rule9.RuleSetModelInstanceRule.BuildingRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    each_rule=Section6Rule9.RuleSetModelInstanceRule.BuildingRule.ZoneRule(),
+                    index_rmr="proposed",
+                    list_path="$..zones[*]",
+                )
+
+            def is_applicable(self, context, data=None):
+                building_p = context.proposed
+                return pint_sum(find_all("spaces[*].floor_area", building_p), ZERO.AREA) >= FLOOR_AREA_LIMIT
+
+            def create_data(self, context, data=None):
+                building_p = context.proposed
+                schedules_p = data["schedules_p"]
+                return {
+                    **data,
+                    "building_open_schedule_p":  getattr_(
+                        find_one_with_field_value("$", "id", building_p["building_open_schedule"], schedules_p),
```

Comment:
As with Rule6-8, I think we need to use `find_exactly_one_with_field_value()` here.